### PR TITLE
Add remote processor firmware loading test (New)

### PR DESCRIPTION
**Pull request by stanley31huang**
## Description
add a new test case to reload the firmware to Remote Processor 

## Resolved issues
N/A

## Documentation
Refer to [TI RemoteProc document](https://software-dl.ti.com/processor-sdk-linux/esd/docs/07_03_00_005/linux/Foundational_Components/PRU-ICSS/Linux_Drivers/RemoteProc_and_RPMsg.html)

## Tests
[ce-oem-rpmsg-automated](https://certification.canonical.com/hardware/202401-33407/submission/366822/)

### CODE REVIEW

**Review by LiaoU3 — CHANGES_REQUESTED** [review]
Please see my suggestions.

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/rpmsg_load_firmware.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+            raise SystemExit("# Reload M4 firmware failed")
+
+        # AI: will adding a feature to do testing after load M-core firmware
+        # extra_testing = None
+        # if extra_testing is not None:
+        #    module_c = __import__(os.path.splitext(__file__)[0])
+        #    getattr(module_c, extra_testing)()
+
+
+def dump_firmware_test_mapping(args):
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/rpmsg_load_firmware.py:182]:
```suggestion
def dump_firmware_test_mapping(args) -> None:
```

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/rpmsg_load_firmware.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                continue
+            for entry in self.log_reader:
+                logging.debug(entry["MESSAGE"])
+                if entry["MESSAGE"] == "":
+                    continue
+                if lookup_func(entry) is False:
+                    return self.expected_events
+
+            cur_time = time.time()
+            if (cur_time - start_time) > 60:
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/rpmsg_load_firmware.py:116]: May I know the reason the delta is 60?

**Review by baconYao — CHANGES_REQUESTED** [review]
Please see my suggestions

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/rpmsg_load_firmware.py (hunk 2 above)
Comment (stanley31huang) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/rpmsg_load_firmware.py:133]: In typical scenarios, the reload firmware test is expected to conclude within 10 seconds. However, given the possibility of executing these tests on systems with lower performance, I have extended the timeout to 60 seconds.

**Review by LiaoU3 — APPROVED** [review]
+1

**Review by LiaoU3 — APPROVED** [review]
+1

**Review by baconYao — APPROVED** [review]
LGTM

**Pull request merged**